rdmd.d

```diff
@@ -321,6 +311,28 @@ int main(string[] args)
         }
     }
 
+    if(makeDepend || makeDepFile !is null)
+    {
+        auto resolvedDeps = (initialDeps is null) ? cast(const(string[string]))readDepsFile(objDir, depsFilename) : initialDeps;
```

Comment:
How about letting `tryGetDependencies` and `readDepsFile` return a `const(string[string]` instead of the "ugly" cast?

Reply:
Wouldn't that just move the cast to the tryGetDependencies and readDepsFile functions?

Reply:
It works for me:

```diff
diff --git a/rdmd.d b/rdmd.d
index 3bc7460..3e17fbd 100755
--- a/rdmd.d
+++ b/rdmd.d
@@ -313,7 +313,8 @@ int main(string[] args)
 
     if(makeDepend || makeDepFile !is null)
     {
-        auto resolvedDeps = (initialDeps is null) ? cast(const(string[string]))readDepsFile(objDir, depsFilename) : initialDeps;
+        const k = initialDeps;
+        auto resolvedDeps = (initialDeps is null) ? readDepsFile(objDir, depsFilename) : k;
         assert(resolvedDeps !is null, "code bug: dependencies were not properly generated from the build");
 
         // --makedepend mode. Just print dependencies and exit.
@@ -610,7 +611,7 @@ private int exec(string[] args)
     return run(args, null, true);
 }
 
-string[string] readDepsFile(string objDir, string depsFilename)
+const(string[string]) readDepsFile(string objDir, string depsFilename)
 {
     string d2obj(string dfile)
     {
@@ -705,7 +706,7 @@ string[string] readDepsFile(string objDir, string depsFilename)
 // directory workDir. The mapping is obtained by reading the 'rdmd.deps' file
 // that would have been generated on a previous build.
 
-private string[string] tryGetDependencies(string rootModule,
+private const(string[string]) tryGetDependencies(string rootModule,
         string objDir, string[] compilerFlags, string depsFilename)
 {
     // Check if the old dependency file is fine
```